src/swarm-topography/Swarm/Game/Scenario/Topography/Structure.hs

```diff
@@ -100,19 +100,26 @@ instance FromJSON (Grid Char) where
           fail "Grid is not rectangular!"
         return g
 
+genStructure ::
```

Comment:
Does the `gen` stand for `generic`?  At first I thought it meant `generate` which didn't make much sense.

Reply:
Renamed to `parseStructure`.